executor/src/witgen/jit/block_machine_processor.rs

```diff
@@ -108,22 +119,123 @@ impl<'a, T: FieldElement> BlockMachineProcessor<'a, T> {
             }
         }
 
-        // If any machine call could not be completed, that's bad because machine calls typically have side effects.
-        // So, the underlying lookup / permutation / bus argument likely does not hold.
-        // TODO: This assumes a rectangular block shape.
-        let has_incomplete_machine_calls = (0..self.block_size)
-            .flat_map(|row| {
-                self.machine_parts
-                    .identities
+        self.check_block_shape(witgen)?;
```

Comment:
I considered doing `unwrap()` here, because for valid block machines, this should never fail.

But it currently fails in one test:
```
--- STDERR:              powdr-pipeline::powdr_std arith256_memory_large_test ---
[ERROR powdr_executor::witgen::jit::block_machine_processor] Column main_arith::carry_high[0] is not stackable:
    ... ??? | <known>, ???, ???, ???, ???, ???, ???, ???, ???, ???, ???, ???, ???, ???, ???, ???, ???, ???, ???, ???, ???, ???, ???, ???, ???, ???, ???, ???, ???, ???, ???, ??? | <known> ...
```

Reply:
Interesting to see the generated code in this example:
```
...
main_arith::carry_high[0][0] = (2147483648 // 65536);
...
main_arith::carry_high[0][32] = (2147483648 // 65536);
```

`hex(2147483648 // 65536) == '0x8000'`. So apparently, the value can be derived at compile time, but it is not represented as a "concrete" value.